rest/route/annotations.go

```diff
@@ -264,6 +265,12 @@ func (h *annotationByTaskPutHandler) Parse(ctx context.Context, r *http.Request)
 			StatusCode: http.StatusBadRequest,
 		}
 	}
+	if t.Status != evergreen.TaskFailed {
```

Comment:
Unlike patch statuses that are limited to failed, succeeded, created, and running, task statuses can have more statuses (like system-failed). I think it might be better to instead check if the task status is equal to success or created (and maybe running depending on David's response) and error if it equals that.

Reply:
actually, scratch that. A better approach would be to use the IsFailedTaskStatus() function.

Reply:
In my mind, anything that is not a success should be annotatable. I'm not sure how evergreen internally represents all the states, but it would be nice to be able to annotate timeouts, system failures, system timeouts and any other abnormal states as well as normal test failures.

Reply:
And how about tasks that haven't finished running yet (and don't have a final status yet, meaning they may end up being success)?

Reply:
We may want to annotate a previous execution of a task that is currently running (execution 2 is in progress, we want to annotate execution 1). But I can't think of a reason why we would want to annotate a task that has not completed yet.

Reply:
done